packages/fondue/src/components/TextInput/TextInput.tsx

```diff
@@ -125,246 +126,261 @@ const ExtraActionButton: ForwardRefRenderFunction<HTMLButtonElement | null, Extr
-    selectable = false,
-    extraActions,
-    maxLength,
-    ...props
-}: TextInputProps): ReactElement => {
-    const { isFocusVisible: inputIsFocusVisible, focusProps: inputFocusProps } = useFocusRing({ isTextInput: true });
-    const { isFocusVisible: clearButtonIsFocusVisible, focusProps: clearButtonFocusProps } = useFocusRing();
-    const { isFocusVisible: passwordButtonIsFocusVisible, focusProps: passwordButtonFocusProps } = useFocusRing();
-    const { isFocusVisible: copyButtonIsFocusVisible, focusProps: copyButtonFocusProps } = useFocusRing();
+export const TextInput = forwardRef<HTMLInputElement, TextInputProps>(
+    (
+        {
+            id: propId,
+            type = TextInputType.Text,
+            decorator,
+            validation = Validation.Default,
+            clearable = false,
+            placeholder,
+            required,
+            obfuscated,
+            disabled = false,
+            autocomplete = false,
+            dotted = false,
+            value = '',
+            copyable = false,
+            onChange,
+            onEnterPressed,
+            onBlur,
+            onClear,
+            onFocus,
+            size,
+            spellcheck,
+            readonly,
+            focusOnMount,
+            selectable = false,
+            extraActions,
+            maxLength,
+            ...props
+        },
+        ref,
+    ): ReactElement => {
+        const inputElement = useRef<HTMLInputElement | null>(null);
+        useImperativeHandle(ref, () => inputElement.current!, []);
 
-    const { copy, status } = useCopy();
+        const { isFocusVisible: inputIsFocusVisible, focusProps: inputFocusProps } = useFocusRing({
+            isTextInput: true,
+        });
+        const { isFocusVisible: clearButtonIsFocusVisible, focusProps: clearButtonFocusProps } = useFocusRing();
+        const { isFocusVisible: passwordButtonIsFocusVisible, focusProps: passwordButtonFocusProps } = useFocusRing();
+        const { isFocusVisible: copyButtonIsFocusVisible, focusProps: copyButtonFocusProps } = useFocusRing();
 
-    const inputElement = useRef<HTMLInputElement | null>(null);
-    const [isObfuscated, setIsObfuscated] = useState(
-        typeof obfuscated === 'boolean' ? obfuscated : type === TextInputType.Password,
-    );
+        const { copy, status } = useCopy();
 
-    useEffect(() => {
-        setTimeout(() => {
-            focusOnMount && inputElement.current?.focus();
-        }, 0);
-    }, [focusOnMount]);
+        const [isObfuscated, setIsObfuscated] = useState(
+            typeof obfuscated === 'boolean' ? obfuscated : type === TextInputType.Password,
+        );
 
-    useEffect(() => {
-        if (typeof obfuscated === 'boolean') {
-            setIsObfuscated(obfuscated);
-        }
-    }, [obfuscated]);
+        useEffect(() => {
+            setTimeout(() => {
+                focusOnMount && inputElement.current?.focus();
+            }, 0);
+        }, [focusOnMount]);
```

Comment:
We don't need this logic here if we expose the entire `ref` (or even just the `focus` method). 

The clients can decide how to manage the focus themselves (on mount, on action, etc). Basically this `useEffect` should live in the client's code:

```ts
import { Button } from '@frontify/fondue';

// the rest of the code [...]

const fondueBtnRef = useRef();

useEffect(() => {
 if (fondueBtnRef.current) {
   fondueBtnRef.current.focus();
}, []);


return <Button ref={fondueBtnRef}>Click Me</Button>
```

Reply:
nevermind, it was a glitch in the GitHub diff, this code was already there and was not changed!

Reply:
Just to clarify, should  I just remove the 'focusOnMount' prop?
I agree to this option. this is a breaking change in web-app and a lot of times needed?

Reply:
yep we should keep it to avoid breaking changes!